src/v/net/connection_rate_counter.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "seastarx.h"
+
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/semaphore.hh>
+#include <seastar/util/bool_class.hh>
+
+using namespace std::chrono_literals;
+
+namespace net {
+
+using should_spawn_fiber_on_update
+  = ss::bool_class<struct should_spawn_fiber_on_update_tag>;
+
+template<typename Clock = ss::lowres_clock>
```

Comment:
@VadimPlh please remember to split your commits up. in particular, changing code (e.g. making this a template) in the same commit that also moves that code to a new file makes reviewing challenging.